test/8_1_CustomTokens.spec.js

```diff
@@ -0,0 +1,270 @@
+      try {
+        await token.addToProjects(address, {from: secondary});
+      } catch ({message}) {
+        error = true;
+        assert.strictEqual(message, getErrorMessage("Ownable: caller is not the owner"))
+      }
+      assert.strictEqual(error, true);
+    });
+
+
+  });
+
+  describe('removeFromProjects()', () => {
+    it('should remove address from projects list', async () => {
+      await token.addToProjects(zeroOne.address);
+      let isProject = await token.isProjectAddress(zeroOne.address);
+      assert.strictEqual(isProject, true);
+
+      await zeroOne.disableUserGroup(token.address);
+      isProject = await token.isProjectAddress(zeroOne.address);
+      assert.strictEqual(isProject, false);
+    });
+
+    it('should fail on removing address from projects list, if sender not project', async () => {
+      await token.addToProjects(zeroOne.address);
+      let isProject = await token.isProjectAddress(zeroOne.address);
+      let error = false;
+      assert.strictEqual(isProject, true);
+      try {
+        await token.removeFromProjects(zeroOne.address, {from});
+        isProject = await token.isProjectAddress(zeroOne.address);
+        console.log(`isproject = ${isProject}`)
+      } catch ({ message }) {
+        error = true;
+        assert.strictEqual(message, getErrorMessage("Address not contains in projects"));
+      }
+      assert.strictEqual(error, true);
+    });
+  });
+
+  describe('transferFrom()', () => {
+    it('should successfully send tokens between users', async () => {
+      await token.transferFrom(from, secondary, 300);
+      const balance = await token.balanceOf(secondary);
+      assert.strictEqual(balance.toNumber(), 300)
+    });
+
+    it('should successfully lock tokens for project on send vote', async () => {
+      await token.addToProjects(zeroOne.address);
+      await zeroOne.setVote(token.address, from, 1);
+      const isLocked =  await token.isTokenLocked(zeroOne.address, from);
+      assert.strictEqual(isLocked, true);
+    });
+
+    it('should fail on lock tokens for project when balance is zero', async () => {
+      let error = false;
+
+      await token.addToProjects(zeroOne.address);
+      try {
+        await zeroOne.setVote(token.address, secondary, 1);
+      } catch ({ message }) {
+        error = true;
+        assert.strictEqual(message, getErrorMessage('Balance of sender must be greater, then zero'))
+      }
+      assert.strictEqual(error, true);
+    });
+
+    it('should fail on send tokens between users, when TX sender not owner or project', async () => {
+      let error = false;
+      await token.transferFrom(from, secondary, 300);
+      try {
+        await token.transferFrom(from, secondary, 300, {from: secondary});
+      } catch ({ message }) {
+        error = true;
+        assert.strictEqual(message, getErrorMessage('This operation is not allowed for this address'));
+      }
+      assert.strictEqual(error, true);
+    });
```

Comment:
lock tokens for 2 projects in same time
double lock
user balance is less then provided